alacritty/src/input.rs

```diff
@@ -864,6 +867,12 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
             },
             ElementState::Released => *self.ctx.suppress_chars() = false,
         }
+
+        // Reset the cursor blinking cycle when the user types
+        if let Some(timer) = self.ctx.scheduler_mut().get_mut(TimerId::CursorBlinking) {
+            timer.deadline = Instant::now() + Duration::from_millis(500);
+            self.ctx.terminal_mut().cursor_blinking_out = false;
+        }
```

Comment:
Should this really always be done when any key is pressed, or only when a character is received? I'd argue that just pressing Control shouldn't disable cursor blinking.